spark/src/main/scala/org/apache/mahout/sparkbindings/drm/package.scala

```diff
@@ -60,19 +60,27 @@ package object drm {
         val keys = data.map(t => t._1).toArray[K]
         val vectors = data.map(t => t._2).toArray
 
-        val block = if (vectors(0).isDense) {
-          val block = new DenseMatrix(vectors.length, blockncol)
-          var row = 0
-          while (row < vectors.length) {
-            block(row, ::) := vectors(row)
-            row += 1
-          }
+        // create the block by default as sparse.
+        // would probably be better to sample a subset of these
+        // vectors first before creating the entire matrix.
+        // so that we don't have the overhead of creating a full second matrix in
+        // the case that the matrix is not Spars
+        val block = new DenseMatrix(vectors.length, blockncol)
+        var row = 0
+        while (row < vectors.length) {
+          block(row, ::) := vectors(row)
+          row += 1
+        }
+
+        // Test the density of the data. If the matrix does not meet the
+        // requirements for sparsity, convert the Vectors to a dense Matrix.
+        val resBlock = if (isMatrixDense(block)) {
           block
```

Comment:
currently fails this test (when testing samples of the full matrix) with a density threshold of .3 rows/matrix containing,  .30% nonZeroElements/row and a sample size or .25 (with a minimum of one row to test):
```scala
test("DRM blockify sparse -> SRM") {

    val inCoreA = sparse(
      (1, 2, 3),
      0 -> 3 :: 2 -> 5 :: Nil
    )
    val drmA = drmParallelize(inCoreA, numPartitions = 2)

    (inCoreA - drmA.mapBlock() {
      case (keys, block) =>
        if (!block.isInstanceOf[SparseRowMatrix])
          throw new AssertionError("Block must be dense.")
        keys -> block
    }).norm should be < 1e-4
  }
```